docs/specifications/tests/Basic-TP/basic00.md

```diff
@@ -1,63 +1,370 @@
+("_"), if used, is only expected to occure in the start of the label. Those
+restrictions on HYPHEN-MINUS, SOLIDUS and LOW LINE are disregarded in this test
+case and are assumed to be handled in test cases [Syntax01] and [Syntax02].
+
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+* "Valid ASCII" - Set of permitted ASCII characters in table 1 above.
+* "Full Stops" - Set of Full Stops in table 3 above.
+* "Unicode" - The Unicode database.
+
+
+## Summary
+
+Message Tag outputted    | Level    | Arguments | Description of when message tag is outputted
+:------------------------|:---------|:----------|:--------------------------------------------
+B00_NOT_UTF-8_ASCII      | CRITICAL |           | Domain name to be tested is encode in unknown character encoding.
+B00_INITIAL_DOT          | CRITICAL |           | Domain name starts with dot.
+B00_REPEATED_DOTS        | CRITICAL |           | Domain name has repeated dots.
+B00_INVALID_ASCII        | CRITICAL | dlabel    | ASCII label has characters not permitted.
+B00_INVALID_U_LABEL      | CRITICAL | dlabel    | Label with non-ASCII character is not valid U-label.
+B00_LABEL_TOO_LONG       | CRITICAL | dlabel    | Domain name has a label that is too long (more than 63 characters).
+B00_DOMAIN_NAME_TOO_LONG | CRITICAL |           | Domain name is too long (more than 253 characters with no final dot).
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+
+## Test procedure
+
+1.  Create an empty, ordered list of labels ("Child Labels").
+
+2.  If *Child Zone* is not encoded in UTF-8 (ASCII is a subset of UTF-8) then
+    output *[B00_NOT_UTF-8_ASCII]* and terminate this test case.
+
+3.  Replace all instances of dots from *Full Stops* in *Child Zone* with U+002E
+    (dot ".").
+
+4.  If *Child Zone* is the root zone, i.e. the exact string ".", then terminate
+    this test case.
+
+5.  If *Child Zone* starts with dot (".") then output *[B00_INITIAL_DOT]* and
+    terminate this test case.
+
+6.  If *Child Zone* has any instance of two or more consecutive dots (".") then
+    output *[B00_REPEATED_DOTS]* and terminate this test.
+
+7.  Remove trailing dot (".") from *Child Zone*.
+
+8.  Split *Child Zone* into labels by dot "." and put them in the same order in
+    *Child Labels*.
+
+9.  For each "Label" in *Child Labels* do:
+    1. If all characters in *Label* are ASCII characters, then do:
+       1. If any character in *Label* is not listed in *Valid ASCII*, then output
+          [B00_INVALID_ASCII] and *Label*, and terminate this test case.
+       2. Else, downcase all uppcase characters as described in section
```

Comment:
souldn't it be "uppercase" instead of "uppcase"?

Reply:
Yes. Fixed.